app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -524,45 +562,79 @@ public void verifyLocationPermissions(Activity activity) {
         int permission = ActivityCompat.checkSelfPermission(activity, Manifest.permission.ACCESS_FINE_LOCATION);
         if (permission != PackageManager.PERMISSION_GRANTED) {
             permissionDenied = true;
-            Log.d("permTag","verify: permissionDenied");
+            Log.d("permTag", "verify: permissionDenied");
             // We don't have permission so prompt the user
             ActivityCompat.requestPermissions(activity, PERMISSIONS_LOCATION, LOCATION_PERMISSION_REQUEST_CODE);
-        }else{
+        } else {
             permissionDenied = false;
         }
     }
 
-    /*
-    // get the selected dropdown list value
-    public void addListenerOnSubmitButton() {
+    public void addItemsOnEmojiScroller() throws IllegalAccessException {
+        Field[] fields = R.drawable.class.getFields();
+        List<Integer> drawables = new ArrayList<Integer>();
+        for (Field field : fields) {
+            // Take only those with name starting with "emoji"
+            if (field.getName().startsWith("emoji")) {
+                drawables.add(field.getInt(null));
+            }
+        }
 
-        //emotionalStateSpinner = (Spinner) findViewById(R.id.emotional_state_spinner);
-        socialSituationSpinner = (Spinner) findViewById(R.id.social_event_spinner);
-        //Button submitButton = (Button) findViewById(R.id.post_mood_button);
-
-        submitButton.setOnClickListener(new View.OnClickListener() {
-
-            @Override
-            public void onClick(View v) {
-
-                emotionalState = String.valueOf(emotionalStateSpinner.getSelectedItem());
-                socialSit = String.valueOf(socialSituationSpinner.getSelectedItem());
-
-                try {
-                    submitMood();
-                    if(editflag) {
-                        Toast.makeText(getApplicationContext(), "Successfully updated!", Toast.LENGTH_SHORT).show();
-                    }
-                    else{
-                        Toast.makeText(getApplicationContext(), "Successfully posted!", Toast.LENGTH_SHORT).show();
-                    }
-                    finish();
-                } catch (DescriptionTooLongException e) {
-                    Toast.makeText(getApplicationContext(), "Your description is too long.", Toast.LENGTH_SHORT).show();
+        emojiList = (LinearLayout) findViewById(R.id.emojiList);
+        for (int i = 1; i <= drawables.size(); i++) {
+
+            LinearLayout emojiLayout = new LinearLayout(this);
+            LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT);
+            params.gravity = Gravity.CENTER_VERTICAL;
+            emojiLayout.setPadding(0,0,20,0);
+            emojiLayout.setOrientation(LinearLayout.VERTICAL);
+            emojiLayout.setTag(i); // This is crucial for grabbing the "index" of the clicked emoji later
```

Comment:
Create a LinearLayout that will hold both an ImageButton and a TextView. This allows us to insert this LinearLayout into the already created LinearLayout within the xml file. The result is a correctly formatted "3D graph" of our scrollbar.